src/ApiProblem.php

```diff
@@ -431,12 +431,24 @@ public function asXml(bool $pretty = false)
      *
      * @return array
      *   The API problem represented as an array.
-    */
+     */
     public function asArray() : array
     {
         return $this->compile();
     }
 
+    /**
+     * Supports rendering this problem as a JSON using the
+     * json_encode() function.
+     *
+     * @return array
+     *   The API problem represented as an array for rendering.
+     */
+    public function jsonSerialize()
+    {
+        return $this->compile();
```

Comment:
This should wrap asJson(), not compile().  asJson() provides some additional error handling we probably want to keep.

Reply:
I'm not sure this would be possible. `asJson()` returns a string formatted according to the options you chose.

`jsonSerialize()` must return an array containing the data that will be encoded to JSON. The formatting options are out of scope here, since it is up to the external encoder (i.e. `json_encode()`) to perform whatever formatting it is told to perform.

Reply:
Also, [json_encode()](https://www.php.net/manual/en/function.json-encode.php) will always return a string. I'm not sure how it would every return `false`.

Reply:
Ah, you're right.  I haven't used JsonSerialize in a very long time so I didn't recall what its contract was.